micronaut-camunda-bpm-feature/src/test/kotlin/info/novatec/micronaut/camunda/bpm/feature/test/MnAdminUserConfigurationTest.kt

```diff
@@ -0,0 +1,137 @@
+package info.novatec.micronaut.camunda.bpm.feature.test
+
+import info.novatec.micronaut.camunda.bpm.feature.Configuration
+import javax.inject.Inject
+import info.novatec.micronaut.camunda.bpm.feature.MnAdminUserConfiguration
+import org.camunda.bpm.engine.ProcessEngine
+import io.micronaut.context.ApplicationContext
+import io.micronaut.core.value.PropertyNotFoundException
+import io.micronaut.test.extensions.junit5.annotation.MicronautTest
+import org.camunda.bpm.engine.authorization.Groups
+import org.camunda.bpm.engine.authorization.Authorization
+import org.camunda.bpm.engine.authorization.Resources
+import org.camunda.bpm.engine.identity.User
+import org.junit.jupiter.api.Assertions
+import org.junit.jupiter.api.Nested
+import org.junit.jupiter.api.Test
+import java.util.*
+
+/**
+ * Tests for [info.novatec.micronaut.camunda.bpm.feature.MnAdminUserConfiguration]
+ *
+ * @author Titus Meyer
+ */
+@MicronautTest(rebuildContext = true)
+internal class MnAdminUserConfigurationTest {
+    @Inject
+    lateinit var configuration: Configuration
+
+    @Inject
+    lateinit var mnAdminUserConfiguration: Optional<MnAdminUserConfiguration>
+
+    @Test
+    fun noAdminUserDefined() {
+        Assertions.assertNotNull(configuration)
+        Assertions.assertThrows(PropertyNotFoundException::class.java) { configuration.adminUser.id }
+        Assertions.assertFalse(mnAdminUserConfiguration.isPresent)
+    }
+
+    @MicronautTest(propertySources = ["classpath:adminusertest.yml"], rebuildContext = true)
+    @Nested
+    internal inner class MnAdminUserConfigurationTestFull {
+        @Inject
+        lateinit var processEngine: ProcessEngine
+
+        @Inject
+        lateinit var configuration: Configuration
+
+        @Inject
+        lateinit var mnAdminUserConfiguration: Optional<MnAdminUserConfiguration>
+
+        @Test
+        fun adminUserCreated() {
+            Assertions.assertNotNull(processEngine)
+            Assertions.assertNotNull(configuration)
+            Assertions.assertTrue(mnAdminUserConfiguration.isPresent)
+
+            Assertions.assertEquals(USERID_VALUE, configuration.adminUser.id)
+            Assertions.assertEquals(PASSWORD_VALUE, configuration.adminUser.password)
+            Assertions.assertEquals(FIRSTNAME_VALUE, configuration.adminUser.firstname)
+            Assertions.assertEquals(LASTNAME_VALUE, configuration.adminUser.lastname)
+
+            assertAdminUser(processEngine, configuration.adminUser.id)
+            assertAdminGroup(processEngine)
+            assertAdminGroupAuthorizations(processEngine)
+        }
+
+        @Test
+        fun adminUserExists() {
+            val properties: MutableMap<String, Any> = HashMap()
```

Comment:
I don't understand this test. Why not simply create a user with the ID and then check in the end that it is still the only same user, i.e. name, firstname, etc are not modified.